python-threatexchange/threatexchange/exchanges/impl/techagainstterrorism_api.py

```diff
@@ -112,13 +108,15 @@ def fetch_iter(
 
         for result in client.fetch_hashes_iter(start_time):
 
-            if result.next is None:
-                return
+            if result.checkpoint:
+                checkpoint = TATCheckpoint(result.checkpoint)
+            else:
+                checkpoint = TATCheckpoint("")
```

Comment:
What cases can cause the checkpoint on the return to be empty? Is the a valid response?

I'm worried that you can accidentally rewind your checkpoint back to the beginning, which is exactly the behavior you were observing earlier on in this, and so I think we should defend against it

If it's not a valid response, then this is a bit less spooky:
```suggestion
            asset result.checkpoint, "API didn't return a checkpoint?"
            # alt: assert not checkpoint or result.checkpoint, "API rewound checkpoint!"
            checkpoint = TATCheckpoint(result.checkpoint)
```

Reply:
Okay,

So if we reach the end of the list and there are no hashes remaining the results list will be `[]` and the checkpoint will be 'null'

and yes you were right it does indeed cause a rewind 🤦